assets/js/editor/Editor.tsx

```diff
@@ -1,55 +1,59 @@
 import React, { useState, useCallback, useEffect, useRef } from 'react';
-import Monaco from "@monaco-editor/react";
-import type { EditorProps as MonacoProps } from  "@monaco-editor/react/lib/types";
+import Monaco from '@monaco-editor/react';
+import type { EditorProps as MonacoProps } from '@monaco-editor/react/lib/types';
 
 import { fetchDTSListing, fetchFile } from '@openfn/describe-package';
 
-const DEFAULT_TEXT = '// Get started by adding operations from the API reference';
+const DEFAULT_TEXT =
+  '// Get started by adding operations from the API reference';
 
 type EditorProps = {
   source?: string;
   adaptor?: string; // fully specified adaptor name - <id>@<version>
   onChange?: (newSource: string) => void;
-}
+  disabled?: string;
```

Comment:
Please change this to a Boolean, there's no reason we can't enforce a boolean on the hook that mounts this component.